core/src/main/java/org/opensearch/sql/planner/logical/LogicalML.java

```diff
@@ -0,0 +1,33 @@
+package org.opensearch.sql.planner.logical;
+
+import java.util.Collections;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.opensearch.sql.ast.expression.Literal;
+
+/**
+ * ML logical plan.
+ */
+@Getter
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class LogicalML extends LogicalPlan {
```

Comment:
Do you plan deprecated Kmeans/AD in Logical plan?

Reply:
Yes, we will deprecate them in the following version like 2.5, then remove them in 3.0.